### PR TITLE
Add part-year-profit-tax-credits Smart Answer

### PR DESCRIPTION
**Pull request by chrisroos**

### CODE REVIEW

**Comment by chrisroos** [comment]
I've rebased this on master and force pushed to this branch.

**Comment by chrisroos** [comment]
I've just rebased on master and force pushed to this branch.

**Comment by chrisroos** [comment]
We've heard back from HMRC and they're pretty much happy with this new Smart Answer so this can now be reviewed.

**Comment by chrisroos** [comment]
I've tweaked the commit titled "Update part-year-profit-tax-credits outcome" based on some more feedback from Gill.

I've rebased on master and force pushed to this branch.

**Comment by floehopper** [comment]
Apart from my minor comments, this looks great! I think the result is a big improvement on existing smart answers and the testing approach seems to work really well. :smile: 

The git history is a little hard to follow in places, but I think that's an inevitable consequence of the amount of iteration that's gone on to get this to where it is. 

I'm happy for it to be merged :+1:

**Comment by chrisroos** [comment]
Thanks for reviewing this @floehopper! I've addressed all your comments and will now rebase and get it merged to master.
